R/compare_df_cols.R

```diff
@@ -0,0 +1,266 @@
+#' @param class_colname The name for the column-name-defining column
+#' @param strict_description Passed to \code{describe_class}
+#' @return A 2-column data.frame with the first column naming all the columns of
+#'   \code{x} and the second column (named after the value in
+#'   \code{class_colname}) defining the classes using
+#'   \code{describe_class()}.
+#' @noRd
+compare_df_cols_df_maker <- function(x, class_colname="class", strict_description)
+  UseMethod("compare_df_cols_df_maker")
+
+compare_df_cols_df_maker.data.frame <- function(x, class_colname="class", strict_description) {
+  if (class_colname == "column_name") {
+    stop('`class_colname` cannot be "column_name"')
+  }
+  if (ncol(x) == 0) {
+    warning(class_colname, " has zero columns and will not appear in output.")
+    ret <- data.frame(column_name=character(0), stringsAsFactors=FALSE)
+  } else {
+    ret <-
+      data.frame(
+        column_name=names(x),
+        X=sapply(X=x, FUN=describe_class, strict_description=strict_description),
+        stringsAsFactors=FALSE
+      )
+    names(ret)[2] <- class_colname
+  }
+  ret
+}
+
+compare_df_cols_df_maker.list <- function(x, class_colname="class", strict_description=strict_description) {
+  if (length(class_colname) != length(x)) {
+    stop("`x` and `class_colname` must be the same length.")
+  } else if (any(class_colname == "column_name")) {
+    stop('`class_colname` cannot be "column_name"')
+  }
+  ret <-
+    lapply(
+      X=seq_along(x),
+      FUN=function(idx) {
+        compare_df_cols_df_maker(x=x[[idx]], class_colname=class_colname[[idx]], strict_description=strict_description)
+      }
+    )
+  Reduce(f=function(x, y) {merge(x, y, by="column_name", all=TRUE)}, x=ret)
+}
+
+#' Are the data.frames the same?
+#'
+#' @inheritParams compare_df_cols
+#' @param verbose Print the mismatching columns if binding will fail.
+#' @return \code{TRUE} if row binding will succeed or \code{FALSE} if it will
+#'   fail.
+#' @family Data frame type comparison
+#' @examples
+#' compare_df_cols_same(data.frame(A=1), data.frame(A=2))
+#' compare_df_cols_same(data.frame(A=1), data.frame(B=2))
+#' compare_df_cols_same(data.frame(A=1), data.frame(B=2), verbose=FALSE)
+#' compare_df_cols_same(data.frame(A=1), data.frame(B=2), bind_method="bind_rows")
+#' @export
+compare_df_cols_same <- function(..., return="mismatch", bind_method=c("rbind", "bind_rows"), verbose=TRUE) {
+  return <- match.arg(return)
+  bind_method <- match.arg(bind_method)
+  ret <- compare_df_cols(..., return=return, bind_method=bind_method)
+  if (nrow(ret) & verbose) {
+    print(ret)
+  }
+  nrow(ret) == 0
+}
+
+#' Describe the class(es) of an object
+#'
+#' @details For package developers, an S3 generic method can be written for
+#'   \code{describe_class()} for custom classes that may need more definition
+#'   than the default method.
+#'
+#' @param x The object to describe
+#' @param strict_description Should the 
```

Comment:
"Should the" - then this trails off

Reply:
I don't know what to put here.  I think I'll merge it - it's the last thing holding it up! - then play with the function; it'll be a test of my understanding it, to be able to figure out what you were going to write here 😆